api/v1alpha3/webhooks.go

```diff
@@ -38,7 +38,7 @@ func aggregateObjErrors(gk schema.GroupKind, name string, allErrs field.ErrorLis
 func isValidSSHKey(sshKey *string) field.ErrorList {
 	var allErrs field.ErrorList
 	if sshKey != nil {
-		reg, err := regexp.Compile("[^-A-Za-z0-9-]+")
+		reg, err := regexp.Compile("[^-A-Za-z0-9_-]+")
```

Comment:
```suggestion
		reg, err := regexp.Compile("[\x00-\x7F]+")
```

Also, can we move this into variable using `regexp.MustCompile`, so we're not reallocating memory on every call to the webhook?

Reply:
If I do `reg := regexp.MustCompile(`[\x00-\x7F]+`)` the tests with ssh key name `test-key`, `test_key`, `test` fails
`

Reply:
I have done some changes. with this change `test\t` is a valid name of ssh keyfile.Hence had to make some changes in the test as well. PTAL